src/NuGet.Clients/NuGet.VisualStudio.OnlineEnvironment.Client/RestoreCommandHandler.cs

```diff
@@ -0,0 +1,106 @@
+        }
+
+        public void RunSolutionRestore()
+        {
+            _joinableTaskFactory.RunAsync(async () =>
+            {
+                var cancellationTokenSource = new CancellationTokenSource();
+                _cancelBuildToken = cancellationTokenSource;
+
+                try
+                {
+                    await GetAndActivatePackageManagerOutputWindowAsync();
+
+                    await TaskScheduler.Default;
+                    IBrokeredServiceContainer serviceContainer = await _asyncServiceProvider.GetServiceAsync<SVsBrokeredServiceContainer, IBrokeredServiceContainer>();
+                    IServiceBroker serviceBroker = serviceContainer.GetFullAccessServiceBroker();
+
+                    var nugetSolutionService = await serviceBroker.GetProxyAsync<INuGetSolutionService>(NuGetServices.NuGetSolutionService);
+
+                    try
+                    {
+                        await nugetSolutionService.RestoreSolutionAsync(cancellationTokenSource.Token);
+                    }
+                    finally
+                    {
+                        (nugetSolutionService as IDisposable)?.Dispose();
+                    }
+                }
+                catch (Exception e)
+                {
+                    // Only log to the activity log for now
+                    ActivityLog.LogError("NuGet Package Manager", e.Message);
+                }
+                finally
+                {
+                    cancellationTokenSource.Dispose();
+                    _cancelBuildToken = null;
+                }
+            });
+        }
+
+        public bool IsRestoreActionInProgress()
+        {
+            return _cancelBuildToken != null;
+        }
+
+        public async Task GetAndActivatePackageManagerOutputWindowAsync()
+        {
+            var outputWindow = await _asyncServiceProvider.GetServiceAsync<SVsOutputWindow, IVsOutputWindow>();
+            var vsUIShell = await _asyncServiceProvider.GetServiceAsync<SVsUIShell, IVsUIShell>();
+
+            await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+
+            // create the Package Manager pane within the Output window
+            var hr = outputWindow.CreatePane(
+                ref GuidList.GuidNuGetOutputWindowPaneGuid,
+                Resources.OutputConsolePaneName,
+                fInitVisible: 1,
+                fClearWithSolution: 0);
+            ErrorHandler.ThrowOnFailure(hr);
+
+            IVsOutputWindowPane pane;
+            hr = outputWindow.GetPane(
+                ref GuidList.GuidNuGetOutputWindowPaneGuid,
+                out pane);
+            ErrorHandler.ThrowOnFailure(hr);
+
+            Guid outputToolWindow = VSConstants.StandardToolWindows.Output;
+
+            vsUIShell.FindToolWindow(0,
+                ref outputToolWindow,
+                out var toolWindow);
+            toolWindow?.Show();
+
+            pane.Activate();
```

Comment:
Does it automatically change the filter to "Package Manager" when showing the Output pane?

Reply:
Yeah, that's what this does.